Sources/TSCUtility/Triple.swift

```diff
@@ -135,6 +143,43 @@ public struct Triple: Encodable, Equatable {
         self.abiVersion = abiVersion
     }
 
+    fileprivate static func parseArch(_ string: String) -> Arch? {
+        let candidates: [String:Arch] = [
+            "x86_64h": .x86_64h,
+            "x86_64": .x86_64,
+            "i686": .i686,
+            "powerpc64le": .powerpc64le,
+            "s390x": .s390x,
+            "aarch64": .aarch64,
+            "amd64": .amd64,
+            "armv7": .armv7(core: nil),
+            "armv6": .armv6,
+            "armv5": .armv5,
+            "arm": .arm,
+            "arm64": .arm64,
+            "arm64e": .arm64e,
+            "wasm32": .wasm32,
+        ]
+        if let match = candidates.first(where: { string.hasPrefix($0.key) })?.value {
```

Comment:
Swift's `Dictionary` isn’t ordered, so we will run into random results here by using `first(where:)`.